src/PluginUpdate.php

```diff
@@ -288,4 +293,30 @@ protected function ads_credits_integration(): void {
 		// Set modals as dismissed and notice as not dismissed.
 		update_option( PINTEREST_FOR_WOOCOMMERCE_OPTION_NAME . '_' . UserInteraction::ADS_MODAL_DISMISSED, true, false );
 	}
+
+	/**
+	 * Token update procedure. V3 to V5.
+	 *
+	 * @since 1.4.0
+	 *
+	 * @return void
+	 */
+	protected function token_update(): void {
+		// Update should only happen if the plugin is connected using the V3 token.
+		if ( ! Pinterest_For_Woocommerce()::is_connected() ) {
```

Comment:
> And because we store data between versions under different keys, it is totally possible to have both at the same time while testing. It may explain why it worked for you and did not for me.

Most likely that. I was also changing some keys in the DB directly so maybe this made it work.

Good catch!

> v3 token is stored as [pinterest_for_woocommerce_data][token][access_token], when v5 is [pinterest_for_woocommerce_data][token_data][access_token]
>
> Not sure this is a good decision, but it is what it is =)

I have changed the name to indicate that now we store not only the token but the refresh token and other data as well.

I have updated the procedure. Please retry.